File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var leftPointer = 0;
    var rightPointer = height.length-1;
    var maxVolume = 0;
    
    while (leftPointer < rightPointer) {
        var currVolume = 0;
        if (height[leftPointer] > height[rightPointer]) {
            currVolume = height[rightPointer] * (rightPointer-leftPointer);
            rightPointer--;
        } else {
            currVolume = height[leftPointer] * (rightPointer-leftPointer);
            leftPointer++;
        }
        maxVolume = Math.max(currVolume, maxVolume);
    }
    
    return maxVolume;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        leftPoint = 0
        rightPoint = len(height)-1
        currMax = 0
        
        while leftPoint < rightPoint:
            currWater = min(height[leftPoint], height[rightPoint]) * (rightPoint - leftPoint)
            currMax = max(currWater, currMax)
            if height[leftPoint] <= height[rightPoint]:
                leftPoint += 1
            else:
                rightPoint -= 1
        
        return currMax